src/routes/NewBoard/NewBoard.tsx

```diff
@@ -1,26 +1,118 @@
 import {API} from "api";
 import "routes/NewBoard/NewBoard.scss";
-import {useState} from "react";
+import React, {useRef, useState} from "react";
 import {AccessPolicySelection} from "components/AccessPolicySelection";
 import {AccessPolicy} from "store/features/board/types";
 import {useTranslation} from "react-i18next";
 import {useNavigate} from "react-router";
+import {TextInputLabel} from "components/TextInputLabel";
+import {TextInput} from "components/TextInput";
+import {Button} from "components/Button";
+import {ScrumlrLogo} from "components/ScrumlrLogo";
+
+import {PassphraseModal} from "components/PassphraseDialog/PassphraseModal/PassphraseModal";
+import {Column, importBoard, Note, Participant, Voting} from "store/features";
+import {useAppDispatch} from "store";
 import {columnTemplates} from "./columnTemplates";
-import {TextInputLabel} from "../../components/TextInputLabel";
-import {TextInput} from "../../components/TextInput";
-import {Button} from "../../components/Button";
-import {ScrumlrLogo} from "../../components/ScrumlrLogo";
+import {Toast} from "../../utils/Toast";
 
 export const NewBoard = () => {
   const {t} = useTranslation();
+  const dispatch = useAppDispatch();
   const navigate = useNavigate();
   const [boardName, setBoardName] = useState<string | undefined>();
   const [columnTemplate, setColumnTemplate] = useState<string | undefined>(undefined);
   const [accessPolicy, setAccessPolicy] = useState(0);
   const [passphrase, setPassphrase] = useState("");
   const [extendedConfiguration, setExtendedConfiguration] = useState(false);
+  const [importFile, setImportFileConfiguration] = useState(false);
+  const [showPasswordModal, setShowPasswordModal] = useState(false);
+  const [loadedFile, setFile] = useState<File>();
+  const [completeBoard, setImportBoard] = useState<BoardContent>();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  type BoardContent = {
+    board: {
+      name: string;
+      description?: string;
+      accessPolicy: string;
+      passphrase?: string;
+    };
+    columns: Column[];
+    notes: Note[];
+    participants: Participant;
+    voting: Voting;
+  };
```

Comment:
I think it should be made more clear that this type is for an imported board. Also, it might make sense to move this to the `types` file for boards.